src/pages/index.js

```diff
@@ -49,59 +32,168 @@ const popupCardForm = document.forms.newCardForm;
+const btnEditProfileImage = document.querySelector('.profile-photo-edit__edit-btn');
 
 const containerSelector = document.querySelector('.cards');
 
 const popupProfileForm = document.forms.profileForm;
 const popupPicture = document.querySelector('.popup_picture');
+const popupConfirmDelete = document.querySelector('.popup_confirm-delete');
+const popupEditProfileImage = document.querySelector('.popup_edit-profile-image');
+const popupProfileImageForm = document.forms.profileEditImageForm;
 
-const cardList = new Section(
-  {
-    data: initialCards,
-    renderer: createCard,
-  },
-  containerSelector // .cards
-);
-
-cardList.renderItems();
+let cardList;
 
 const profileFormValidator = new FormValidator(options, popupProfileForm);
 profileFormValidator.enableValidation();
 
 const newCardFormValidator = new FormValidator(options, popupCardForm);
 newCardFormValidator.enableValidation();
 
-function createCard(cardData) {
-  const card = new Card(cardData, '.template-card', handlePicturePopup);
-  const cardElement = card.getCardElement();
-  cardList.addItem(cardElement);
-}
+const ProfileImageFormValidator = new FormValidator(options, popupProfileImageForm);
+ProfileImageFormValidator.enableValidation();
 
 const popupImage = new PopupWithImage(popupPicture);
 
 function handlePicturePopup(name, link) {
   popupImage.open(name, link);
 }
 
-const popupNewCard = new PopupWithForm(newCard, (data) => {
-  createCard(data);
-  popupNewCard.reset();
-  newCardFormValidator.disableSubmitButton();
-});
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    '.template-card',
+    handlePicturePopup,
+    () => {
+      popupDelete.setAction(() => {
+        api
+          .deleteCard(cardData._id)
+          .then(() => {
+            cardElement.remove();
+            popupDelete.close();
+          })
+          .catch((error) => {
+            console.error('Error deleting card:', error);
+          });
+      });
+      popupDelete.open();
+    },
+    handleLikeButton
+  );
+  const cardElement = card.getCardElement();
+  cardList.addItem(cardElement);
+}
+
+const popupDelete = new PopupConfirmation(popupConfirmDelete);
+
+/////
+function handleLikeButton(cardId) {
+  if (this.isLiked()) {
+    api
+      .dislikeCard(cardId)
+      .then((res) => this.setIsLiked(res.isLiked))
+      .catch((error) => {
+        console.error(error);
+      });
+  } else {
+    api
+      .likeCard(cardId)
+      .then((res) => this.setIsLiked(res.isLiked))
+      .catch((error) => {
+        console.error(error);
+      });
+  }
+}
+
+const loadingButtonText = 'Saving...';
+
+const popupNewCard = new PopupWithForm(newCard, handleNewCardSubmition, loadingButtonText);
+
+function handleNewCardSubmition(cardData) {
+  popupNewCard.renderLoading(true);
+  api
+    .addCard(cardData)
+    .then(() => {
+      createCard(cardData);
+    })
+    .finally(() => {
+      popupNewCard.renderLoading(false);
+      popupNewCard.reset();
```

Comment:
Needs correction: you should reset the form, close the popup and disable the button in `then` block, otherwise the form will close  even if an error occurred and you will need to reopen the form and fill the values again.